src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -372,11 +371,17 @@ def create_binary_folder(cls, folder, sorting, recording, sparsity, return_scale
         # NumpyFolderSorting.write_sorting(sorting, folder / "sorting")
         sorting.save(folder=folder / "sorting")
 
-        # save recording and sorting provenance
-        if recording.check_serializability("json"):
-            recording.dump(folder / "recording.json", relative_to=folder)
-        elif recording.check_serializability("pickle"):
-            recording.dump(folder / "recording.pickle", relative_to=folder)
+        if recording is not None:
```

Comment:
we should have a mutualy exclusive between recording and rec_attributes in this function at some point